src/extension/commandPaletteHandler.ts

```diff
@@ -21,32 +22,34 @@ export class CommandPaletteHandler {
     /**
      * Starts the React Native packager
      */
-    public startPackager(): void {
-        this.executeCommandInContext("startPackager", () => this.reactNativePackager.start(vscode.window.createOutputChannel("React-Native")));
+    public startPackager(): Q.Promise<void> {
+        return this.executeCommandInContext("startPackager", () => this.reactNativePackager.start(vscode.window.createOutputChannel("React-Native")));
     }
 
     /**
      * Kills the React Native packager invoked by the extension's packager
      */
-    public stopPackager(): void {
-        this.executeCommandInContext("stopPackager", () => this.reactNativePackager.stop(vscode.window.createOutputChannel("React-Native")));
+    public stopPackager(): Q.Promise<void> {
+        return this.executeCommandInContext("stopPackager", () => this.reactNativePackager.stop(vscode.window.createOutputChannel("React-Native")));
     }
 
     /**
      * Executes the 'react-native run-android' command
      */
-    public runAndroid(): void {
-        this.executeCommandInContext("runAndroid", () => this.executeReactNativeRunCommand("run-android"));
+    public runAndroid(): Q.Promise<void> {
+        return this.executeCommandInContext("runAndroid", () => this.executeReactNativeRunCommand("run-android"));
     }
 
     /**
      * Executes the 'react-native run-ios' command
      */
-    public runIos(): void {
-        // Set the Debugging setting to disabled
-        new IOSDebugModeManager(this.workspaceRoot).setSimulatorJSDebuggingModeSetting(/*enable=*/ false)
-            .catch(() => { }) // If setting the debugging mode fails, we ignore the error and we run the run ios command anyways
-            .done(() => this.executeCommandInContext("runIos", () => this.executeReactNativeRunCommand("run-ios")));
+    public runIos(): Q.Promise<void> {
+        return this.executeCommandInContext("runIos", () => {
+            // Set the Debugging setting to disabled, because in iOS it's persisted across runs of the app
+            return new IOSDebugModeManager(this.workspaceRoot).setSimulatorJSDebuggingModeSetting(/*enable=*/ false)
+                .catch(() => { }) // If setting the debugging mode fails, we ignore the error and we run the run ios command anyways
```

Comment:
Not the way the code is currently set up. Right now if you haven't run the app before, `setSimulatorJSDebuggingModeSetting` will actually reject after 10 seconds or so of trying to find the file, but that is 'working as intended'